internal/webserver/webserver.go

```diff
@@ -355,3 +360,28 @@ func (n kubeFilter) getProxyTenantsForOwnerKind(ownerKind capsulev1beta1.OwnerKi
 
 	return
 }
+
+func (n *kubeFilter) removingHopByHopHeaders(request *http.Request) {
+	connectionHeaderName, upgradeHeaderName, requestUpgradeType := "connection", "upgrade", ""
```

Comment:
These are case sensitive when used for direct header map access.

```suggestion
	connectionHeaderName, upgradeHeaderName, requestUpgradeType := "Connection", "Upgrade", ""
```

Reply:
Or use 'request.Header.Values[connectionHeaderName]'

Reply:
Solved using the case-insensitive values retrieval function.